File: PS1/nheyek2/Diagonal.py
'''
************************************************
  ____ ____    _  ___   __
 / ___/ ___|  / |/ _ \ / /_
| |   \___ \  | | (_) | '_ \
| |___ ___) | | |\__, | (_) |
 \____|____/  |_|  /_/ \___/


Problem set 1


Question 2

Given a square matrix size N x N, calculate the product of the sums across the two main diagonals.
For example, given the input:

3

4   5   7
3   1   5
9   3   2

Return: (4+1+2)*(9+1+7) = 119

PLEASE LOOK AT PS1.txt FOR MORE DETAILS!!!

************************************************
'''

def diagonal(input):
    sum1 = 0
    sum2 = 0
    grid_length = len(input)

    #determines the sum of the first diagonal
    for val in range(0,grid_length):
        sum1 += int(input[val][val])

    #determines the sum of the second diagonal
    for val in range(0,grid_length):
        sum2 += int(input[val][len(input)-1-val])

    product = sum1 * sum2
    return product

try:
    with open('Diagonal.txt') as f:
        data = f.read().split()
        num_of_tests = data[0]
        data = data[1:]

        for i in range(0, int(num_of_tests)):
            matrix = []
            dimen = int(data[0])
            data = data[1:]

            for j in range(0, dimen):
                row = data[:dimen]
                matrix.append(row)
                data = data[dimen:]
            print diagonal(matrix)
except IOError, e:
    print e

File: PS1/tulshib2/Diagonal.java
import java.util.*;
import java.io.*;

/*
************************************************
  ____ ____    _  ___   __   
 / ___/ ___|  / |/ _ \ / /_  
| |   \___ \  | | (_) | '_ \ 
| |___ ___) | | |\__, | (_) |
 \____|____/  |_|  /_/ \___/ 
                             

Problem set 1


Question 2

Given a square matrix size N x N, calculate the product of the sums across the two main diagonals.
For example, given the input:

3

4	5	7
3	1	5
9	3	2

Return: (4+1+2)*(9+1+7) = 119

PLEASE LOOK AT PS1.txt FOR MORE DETAILS!!!

************************************************

*/
public class Diagonal {

	public static int diagonal(int[][] input) throws IOException {
		if(input.length != input[0].length)
			throw new IOException("Matrix is not square!");
		int out = 0, sum1 = 0, sum2 = 0;
		for(int i = 0; i < input.length; i++){
			sum1 += input[i][i];
			sum2 += input[input.length - i - 1][i];
		}
		///DEBUG PRINT
		//System.out.println(sum1 + "\n"+ sum2);
		out = sum1 * sum2;
		return out;
	}





	public static void main(String[] args) {
		File file = new File("Diagonal.txt");
		try {
			Scanner scan = new Scanner(file);
			int numberOfCases = scan.nextInt();
			for(int i = 0; i < numberOfCases; i++) {
				int N = scan.nextInt();
				int[][] matrix = new int[N][N];
				for(int j = 0; j < N; j++) {
					for(int k = 0; k < N; k++) {
						matrix[j][k] = scan.nextInt();
					}
				}
				try{
					System.out.println(diagonal(matrix));
				} catch (IOException e) {
					e.printStackTrace();
				}
			}
			scan.close();
		}
		catch (FileNotFoundException e) {
			e.printStackTrace();
		}
	}

}